server/src/main/java/org/candlepin/resource/StatusResource.java

```diff
@@ -89,17 +96,23 @@ public StatusResource(RulesCurator rulesCurator,
     @Produces({ MediaType.APPLICATION_JSON})
     @SecurityHole(noAuth = true, anon = true)
     public Status status() {
+        /*
+         * Originally this was used to indicate database connectivity being good/bad.
+         * In reality it could never be false, the request would fail. This check has
+         * been moved to GET /status/db.
+         */
         boolean good = true;
         try {
-            if (rulesCurator != null) {
-                rulesCurator.listAll();
-            }
+            rulesCurator.getUpdatedFromDB();
         }
         catch (Exception e) {
+            log.error("Error checking database connection", e);
             good = false;
         }
+
         Status status = new Status(good, version, release, standalone,
```

Comment:
Would it make sense to change this method around to store a status instance and simply return it unless (a) it's null (b) it's older than X seconds or (c) a "force refresh" flag is provided? That would probably cut down some extra processing time a bit, assuming users didn't just always apply the force refresh flag.